### DISCUSSION TITLE
Fixing the dependency warnings.

### DISCUSSION THREAD

**Discussion by EdwardMoyse** [HSF/phoenix#541] [Q&A] [answered]
As I'm sure others have too, I've been noticing many dependency warnings within Phoenix, and I decided to give a go at fixing them. The first thing I decided to do was to update to yarn 2.0 (since apparently this would help).

One example of a warning is:
```
➤ YN0060: │ phoenix-ng@workspace:packages/phoenix-ng provides @angular/common (p2a6d5) with version 15.0.4, which doesn't satisfy what @rp3e11/ngx-slider requests
```

So this I tried to fix with `yarn up @angular/common`

Unfortunately this fails:
```
➤ YN0007: │ root@workspace:. must be built because it never has been before or the last one failed
➤ YN0009: │ root@workspace:. couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-55b22fd2/build.log)
```
  
with:
```
# This file contains the result of Yarn building a package (root@workspace:.)
# Script name: postinstall

lerna notice cli v6.3.0
lerna info Symlinking packages and binaries
src/event-display.ts(3,23): error TS2307: Cannot find module 'jsroot/geom' or its corresponding type declarations.
src/event-display.ts(4,26): error TS2307: Cannot find module 'jsroot/io' or its corresponding type declarations.
```

@9inpachi ... any idea?

**Comment by 9inpachi** [comment]
This is unrelated to the command you shared. It's happening after installing dependencies when `phoenix-event-display` is built. This error shouldn't appear because I already added types to jsroot and linked to my version of the jsroot repo. You can check in the lock file if it's indeed using the correct version `9inpachi/jsroot`.

It's related to jsroot not supporting TypeScript and exposing the public API by specifying modules in `package.json` but it shouldn't be happening since I already fixed it.

**Reply by EdwardMoyse** [reply]
Hi @9inpachi - thanks for your reply! Yes, I didn't mean to imply that the problem was caused by `yarn up`, but that I cannot continue because of it.

Here's a simple recipe to reproduce.

```
git clone git@github.com:HSF/phoenix.git
cd phoenix
npm install -g yarn
yarn set version berry
nano .yarnrc.yml # Add nodeLinker: node-modules in your .yarnrc.yml file
yarn install
```
This will fail with 

```
➤ YN0009: │ root@workspace:. couldn't be built successfully (exit code 1, logs can be found here: /private/var/folders/j6/cv3xx2910nx_pg7v2t8czqzc0000gn/T/xfs-20f41c87/build.log)
```

I confirm that it's being pulled from your version:
```
emoyse@macminim1umass phoenix % cat yarn.lock | grep jsroot
"jsroot@github:9inpachi/jsroot#7.2.0-types":
  resolution: "jsroot@https://github.com/9inpachi/jsroot.git#commit=05f28a38153cf031bef5a334d649dd9fdb8b6c08"
    jsroot: "github:9inpachi/jsroot#7.2.0-types"
```
(incidentally, jsroot 7.3.0 is out now - unfortunately I guess it will still not work for us without changes like you made last time)

**Comment by 9inpachi** [comment, marked as answer]
If you check for `jsroot` in `node_modules`, you will see that the files are not actually the ones from [my repo](https://github.com/9inpachi/jsroot/tree/7.2.0-types). Yarn isn't resolving the dependency correctly.

> jsroot 7.3.0 is out now

Cool. The types were not included in the `7.2.1` version as I asked linev about it [here](https://github.com/root-project/jsroot/pull/246#issuecomment-1332680206). But they have been added in the minor release `7.3.0`. We should be able to switch to the new version.

**Reply by EdwardMoyse** [reply]
Thanks @9inpachi - I opened a ticket about upgrading to yarn 2 (#545 ). I suggest we follow up there.
